src/handlers/rest.ts

```diff
@@ -8,7 +8,17 @@ const createRESTHandler = (method: RESTMethods) => {
       mask,
       predicate(req) {
         const hasSameMethod = method === req.method
-        const urlMatch = match(mask, req.url)
+
+        // Prepends a host origin to the routes that start
+        // with the slash ("/"). This way such routes will match
+        // the respective hostname's routes, while bypassing
+        // the routes from different hosts.
+        const resolvedMask =
+          typeof mask === 'string' && mask.startsWith('/')
+            ? `${location.origin}${mask}`
```

Comment:
I'm not tackling this on the `node-match-path` level, as such a low-level library must not care about the hosts, origins, etc. Therefore, I'm checking if dealing with the route that starts with a slash, and prepending an origin address (protocol + hostname + port, without routes).

Reply:
I've chosen to prepend this information here for a purpose of preserving the routes in the mocking schema as-is. It would be easier for developers to navigate through the schema if their input is left intact.